File: programacao/c/respostas/operacoes/38.c
#include <stdio.h>
#include <math.h>

int main() {
    const double R = 6371.01;
    double x1, y1, x2, y2;
    double d;

    printf("Digite a latitude do primeiro ponto (graus): ");
    scanf("%lf", &x1);

    printf("Digite a longitude do primeiro ponto (graus): ");
    scanf("%lf", &y1);

    printf("Digite a latitude do segundo ponto (graus): ");
    scanf("%lf", &x2);

    printf("Digite a longitude do segundo ponto (graus): ");
    scanf("%lf", &y2);

    x1 = x1 * M_PI / 180;
    y1 = y1 * M_PI / 180;
    x2 = x2 * M_PI / 180;
    y2 = y2 * M_PI / 180;

    d = R * acos(sin(x1) * sin(x2) + cos(x1) * cos(x2) * cos(y1 - y2));

    printf("A distancia entre os pontos e %.2f km\n", d);

    return 0;
}

File: programacao/python/respostas/operacoes/38.py
import math

x1 = float(input("Latitude do primeiro ponto (graus): "))
y1 = float(input("Longitude do primeiro ponto (graus): "))
x2 = float(input("Latitude do segundo ponto (graus): "))
y2 = float(input("Longitude do segundo ponto (graus): "))

x1_rad = math.radians(x1)
y1_rad = math.radians(y1)
x2_rad = math.radians(x2)
y2_rad = math.radians(y2)

R = 6371.01
d = R * math.acos(
    math.sin(x1_rad) * math.sin(x2_rad)
    + math.cos(x1_rad) * math.cos(x2_rad) * math.cos(y1_rad - y2_rad)
)

print(f"A distancia entre os pontos e {d:.2f} km")
